### PR TITLE
Change admin endpoint name

### LINKED ISSUE #3179: Enable serving admin site at custom and secure url
The new admin page looks really, really nice.

But as far as I can see everybody can access it if enabled. I would like to be able to restrict access.

For example via a username and password. Or alternatively at a custom "token" url by specifying something like `--admin=879B996862F69D8561D1B369F7E44` and then the admin page is available at `/879B996862F69D8561D1B369F7E44`.

My use case would be for example awesome-panel.org where I finally could start getting some insights :-)

### LINKED ISSUE #5370: Only allow certain users to see Admin panel
#### Is your feature request related to a problem? Please describe.
I have a user facing app.  With Okta as my Oauth provider.  The admin panel is very useful but I would like only the devs to be able to access it.  As it stands anyone can accidentally stumble into the /admin endpoint

#### Describe the solution you'd like
I may be well off the mark here but my initial thoughts would be if the admin panel could be a pane perhaps?  Then the logic of the code could hide it or show it based on the user info from theauthentication provider

#### Describe alternatives you've considered
Only alternative I have is to use the **debugger** inside the app and have that available to only certain users.  But would love all the other features of the admin panel aswell

#### Additional context
This was just a rough thought.  If this is not even possible I am happy to be told so.  I do plan to build my own metrics and the like in the future.  But I like the functionality of the admin panel as is.

### PR DESCRIPTION
**Pull request by ndmlny-qs**
Resolves #5370 
Resolves #3179

### CODE REVIEW

File: panel/command/serve.py [hunk 1]
```diff
@@ -182,6 +182,12 @@ class Serve(_BkServe):
             action  = 'store_true',
             help    = "Whether to add an admin panel."
         )),
+        ('--admin-panel-name', dict(
+            metavar = "KEY=VALUE",
+            nargs   = '+',
+            help    = "Name to use for the admin panel.",
+            default = '/admin'
+        )),
```
Comment (philippjfr) [inline panel/command/serve.py:190]: I'd call it `--admin-endpoint` for consistency and use `action = 'store', type=str`

Comment (ndmlny-qs) [reply panel/command/serve.py]: done

**Comment by ndmlny-qs** [comment]
Added documentation to `profiling/admin.md` as well.

I'm going to keep this PR in a `[WIP]` state for now. The ability to change the name of the admin endpoint does not prevent other users from seeing it when they are logged in, it just obfuscates where the admin endpoint is. I can't entirely work out the data flow for how the admin page is created, but it doesn't seem to be going through the `DocHandler.get` method, otherwise PR #5386 would prevent a non-authorized user from seeing the renamed admin page.

**Comment by ndmlny-qs** [comment]
switching from `WIP` to a PR

File: panel/command/serve.py [hunk 2]
```diff
@@ -182,6 +182,12 @@ class Serve(_BkServe):
             action  = 'store_true',
             help    = "Whether to add an admin panel."
         )),
+        ('--admin-endpoint', dict(
+            metavar = "KEY=VALUE",
+            nargs   = '+',
```
Comment (philippjfr) [inline panel/command/serve.py:187]:
```suggestion
```

File: panel/command/serve.py [hunk 3]
```diff
@@ -182,6 +182,12 @@ class Serve(_BkServe):
             action  = 'store_true',
             help    = "Whether to add an admin panel."
         )),
+        ('--admin-endpoint', dict(
+            metavar = "KEY=VALUE",
```
Comment (philippjfr) [inline panel/command/serve.py:186]:
```suggestion
        type    = str,
```

File: panel/command/serve.py [hunk 4]
```diff
@@ -182,6 +182,12 @@ class Serve(_BkServe):
             action  = 'store_true',
             help    = "Whether to add an admin panel."
         )),
+        ('--admin-endpoint', dict(
+        action = 'store',
+        type    = str,
```
Comment (philippjfr) [inline panel/command/serve.py:187]:
```suggestion
            action = 'store',
            type    = str,
```

File: panel/command/serve.py [hunk 5]
```diff
@@ -348,17 +354,24 @@ def customize_kwargs(self, args, server_kwargs):
         if args.admin:
             from ..io.admin import admin_panel
             from ..io.server import per_app_patterns
+
+            # NOTE: `admin_endpoint` returns a list. If the command line flag
+            #       `--admin-endpoint` is not used, then we default to the `/admin` path.
```
Comment (philippjfr) [inline panel/command/serve.py:359]:
```suggestion
            # If `--admin-endpoint` is not set, then we default to the `/admin` path.
```

File: panel/command/serve.py [hunk 6]
```diff
@@ -348,17 +354,23 @@ def customize_kwargs(self, args, server_kwargs):
         if args.admin:
             from ..io.admin import admin_panel
             from ..io.server import per_app_patterns
+
+            # If `--admin-endpoint` is not set, then we default to the `/admin` path.
+            admin_path = "/admin"
+            if args.admin_endpoint:
+                admin_path = args.admin_endpoint[0]
```
Comment (philippjfr) [inline panel/command/serve.py:361]:
```suggestion
                admin_path = args.admin_endpoint
```

**Review by philippjfr — APPROVED** [review]
Thank you, took the liberty to make some minor changes and add some tests.

**Pull request merged**